utils/keys.go

```diff
@@ -103,11 +105,73 @@ func ImportKeys(from io.Reader, to []Importer) error {
 		toWrite []byte
 	)
 	for block, rest := pem.Decode(data); block != nil; block, rest = pem.Decode(rest) {
+		// if there is a path then we set the gun header from this path
+		if rawPath := block.Headers["path"]; rawPath != "" {
+			pathWOFileName := strings.TrimSuffix(rawPath, filepath.Base(rawPath))
+			if strings.HasPrefix(pathWOFileName, notary.NonRootKeysSubdir) {
+				gunName := strings.TrimPrefix(pathWOFileName, notary.NonRootKeysSubdir)
```

Comment:
Can we add a test-case that if we initialize a repo, export keys, then import keys to a fresh file-based keystore the `gun` PEM headers are set?

Also, for this logic - it seems like delegation keys may end up with a `gun` header with an empty value?  Should we add a check for `gunName` being an empty string?